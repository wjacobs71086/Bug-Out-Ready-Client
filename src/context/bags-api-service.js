import config from '../config'
import TokenService from './token-service';

const BagsApiService = {
  getThings() {
    return fetch(`${config.API_ENDPOINT}/bags`, {
      headers: {
        'authorization': `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
  },
  getThing(bagId) {
    console.log()
    return fetch(`${config.API_ENDPOINT}/bag-home/${bagId}`, {
      headers: {
        'authorization': `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then(res => {
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      })
  },
  getBagsItems(bagId) {
    console.log(`${config.API_ENDPOINT}/bag-home/${bagId}`);
    return fetch(`${config.API_ENDPOINT}/bag-home/${bagId}`, {
      headers: {
        'authorization': `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
  }

}

export default BagsApiService;