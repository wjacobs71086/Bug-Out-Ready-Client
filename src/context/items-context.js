import React, { Component } from 'react';


const ItemsListContext = React.createContext({
  itemsList: [],
  bagsList: [],
  userId: [],
  error: null,
  setError: () => {},
  clearError: () => {},
  setItemsList: () => {},
  setBagsList: () => {},
})
export default ItemsListContext

export class ItemsListProvider extends Component {
  state = {
    itemsList: [],
    bagsList: [],
    error: null,
  };

  setItemsList = itemsList => {
    this.setState({ itemsList })
  }

  setBagsList = bagsList => {
    this.setState({ bagsList })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render() {
    const value = {
      itemsList: this.state.itemsList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setItemsList: this.setItemsList,
      setBagsList: this.setBagsList,
      bagsList: this.state.bagsList,
      userId: this.state.userId,
    }
    return (
      <ItemsListContext.Provider value={value}>
        {this.props.children}
      </ItemsListContext.Provider>
    )
  }
}