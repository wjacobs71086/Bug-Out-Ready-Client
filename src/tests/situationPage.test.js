import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SituationPage from '../routes/situation-page/situation-page';
import { BrowserRouter } from 'react-router-dom'
import { ItemsListProvider } from '../context/items-context';
import { Route } from 'react-router-dom'

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
            <ItemsListProvider>
                <Route
                    component={SituationPage}
                />
            </ItemsListProvider>
        </BrowserRouter>
        , div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(        
      <BrowserRouter>
        <ItemsListProvider>
            <Route
                component={SituationPage}
            />
        </ItemsListProvider>
    </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
});