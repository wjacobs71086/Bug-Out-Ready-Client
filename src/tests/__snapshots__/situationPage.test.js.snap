// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <img
    alt="bag"
    className="logo"
    src="bags_bag_handbag_accessory_accessories-19-512.png"
  />
  <h3
    className="situationTitle"
  >
    Create a new bag
  </h3>
  <div>
    <form
      className="NewBagForm"
      onSubmit={[Function]}
    >
      <div
        role="alert"
      />
      <div
        className="bag_name"
      >
        <input
          className="Input "
          id="NewBagForm__bag_name"
          name="bag_name"
          placeholder="Bag name"
          required={true}
        />
      </div>
      <label
        className="selectorLabel"
      >
        Bag designed for: 
      </label>
      <select
        className="selector"
        name="situations"
        required={true}
      >
        <option
          value="Everything"
        >
          Everything
        </option>
        <option
          value="Quake"
        >
          Earthquake
        </option>
        <option
          value="Fire"
        >
          Wild Fire
        </option>
        <option
          value="Flood"
        >
          Flash Flood
        </option>
      </select>
      <div
        className="buttons"
      >
        <button
          className="newBagButton"
        >
          <a
            href="/bags"
            id="cancelButton"
            onClick={[Function]}
          >
            Cancel
          </a>
        </button>
        <button
          className="newBagButton"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
